evm_arithmetization/src/witness/util.rs

```diff
@@ -258,16 +262,18 @@ pub(crate) fn stack_pop_with_log_and_fill<const N: usize, F: Field>(
 
 fn xor_into_sponge<F: Field>(
     state: &mut GenerationState<F>,
+    is_generation: bool,
     sponge_state: &mut [u8; KECCAK_WIDTH_BYTES],
     block: &[u8; KECCAK_RATE_BYTES],
 ) {
     for i in (0..KECCAK_RATE_BYTES).step_by(32) {
         let range = i..KECCAK_RATE_BYTES.min(i + 32);
         let lhs = U256::from_little_endian(&sponge_state[range.clone()]);
         let rhs = U256::from_little_endian(&block[range]);
-        state
-            .traces
-            .push_logic(logic::Operation::new(logic::Op::Xor, lhs, rhs));
+        state.traces.push_logic(
+            is_generation,
```

Comment:
nit: `is_generation` isn't straightforward when going through the code. Maybe renaming the bool to `append_to_trace` or something along those lines?